node/src/Pos/Explorer/Txp/Toil/Logic.hs

```diff
@@ -160,6 +163,17 @@ delTxExtraWithHistory id addrs = do
     for_ addrs $ modifyAddrHistory $
         NewestFirst . delete id . getNewestFirst
 
+updateUtxoSumFromBalanceUpdate :: MonadTxExtra m => BalanceUpdate -> m ()
+updateUtxoSumFromBalanceUpdate balanceUpdate = do
+    let plusChange  = foldr unsafeAddCoin (mkCoin 0) $ map snd $ plusBalance  balanceUpdate
+        minusChange = foldr unsafeAddCoin (mkCoin 0) $ map snd $ minusBalance balanceUpdate
+        utxoChange  =
+            if plusChange > minusChange then
+                (Plus, unsafeSubCoin plusChange minusChange)
+            else
+                (Minus, unsafeSubCoin minusChange plusChange)
+    uncurry updateUtxoSum utxoChange
```

Comment:
This treatment of signs here is quite baroque. This is what negative numbers were invented for. Would it not be easier to convert to a signed type and convert back at the end?